lib/gemstash/migrations/03_cached_gems.rb

```diff
@@ -0,0 +1,24 @@
+Sequel.migration do
+  change do
+    create_table :upstreams do
+      primary_key :id
+      String :uri, size: 191, null: false
+      String :host_id, size: 191, null: false
+      DateTime :created_at, null: false
+      DateTime :updated_at, null: false
+      index [:uri], unique: true
+      index [:host_id], unique: true
+    end
+
+    create_table :cached_rubygems do
+      primary_key :id
+      Integer :upstream_id, null: false
+      String :name, size: 191, null: false
+      String :resource_type, size: 191, null: false
+      DateTime :created_at, null: false
+      DateTime :updated_at, null: false
+      index [:upstream_id, :name, :resource_type], unique: true
```

Comment:
I'm thinking that the resource type will change less than the name, could this index be upstream_id, resource_type, name ?

All the variance will be in the name, right? The resource type will be pretty much constant, or we are either going to get the content or something else, not both at the same time.